### PR TITLE
Add loadingComponent prop and minor changes to loading state

### LINKED ISSUE #3: Loading state shouldn't be determined by the presence of loadingMessage
It should ideally be handled by a `loading` boolean prop.

### PR DESCRIPTION
**Pull request by dextermb**
This pull request adds the `loadingComponent` prop, fixes some prop types and controls the loading state by the new `loading` prop rather than checking if there's a loading message.

Closes #3

### CODE REVIEW

File: src/DynamicDataTable.jsx [hunk 1]
```diff
@@ -325,18 +325,38 @@ class DynamicDataTable extends Component {
...
+                                {!!loadingMessage && !!loadingIndicator && (
+                                    <React.Fragment>
+                                        <div className="mt-5">
+                                            { loadingIndicator }
+                                        </div>
+                                        <div className="mt-5">
+                                            { loadingMessage }
+                                        </div>
+                                    </React.Fragment>
+                                )}
```
Comment (DivineOmega) [inline src/DynamicDataTable.jsx]: Could this section just be done as follows?

```jsx
<React.Fragment>
    { loadingIndicator && (
        <div className="mt-5">
            { loadingIndicator }
        </div>
    ) }
    
    { loadingMessage && (
        <div className="mt-5">
            { loadingMessage }
        </div>
    ) }
</React.Fragment>
```

File: src/DynamicDataTable.jsx [hunk 2]
```diff
@@ -325,18 +325,38 @@ class DynamicDataTable extends Component {
     }
 
     renderLoadingTable() {
+        const { loadingIndicator, loadingMessage, loadingComponent } = this.props;
+
+        if (!!loadingComponent) {
```
Comment (DivineOmega) [inline src/DynamicDataTable.jsx]: I'd prefer the following format for readability, unless you can provide a common example where the boolean conversion would be beneficial. Not not just seems quite unclear.

```suggestion
        if (loadingComponent) {
```

File: src/DynamicDataTable.jsx (hunk 1 above)
Comment (DivineOmega) [reply src/DynamicDataTable.jsx]: This would also ensure there is a top margin if only one element is present.

File: src/DynamicDataTable.jsx [hunk 3]
```diff
@@ -427,7 +449,9 @@ DynamicDataTable.defaultProps = {
     orderByDirection: 'asc',
     renderCheckboxes: false,
     actions: [],
-    loadingMessage: '',
+    loading: false,
+    loadingMessage: 'Loading data...',
+    loadingIndicator: null,
```
Comment (DivineOmega) [inline src/DynamicDataTable.jsx]: Can you alter existing documentation to describe these new props?

File: src/DynamicDataTable.jsx (hunk 1 above)
Comment (DivineOmega) [reply src/DynamicDataTable.jsx]: If the loading message looks fine without a top margin, feel free to remove it, and just do the following.

```jsx
<React.Fragment>
    { loadingIndicator ? (<div>{ loadingIndicator }</div>) : null }
    { loadingMessage ? (<div>{ loadingMessage }</div>) : null }
</React.Fragment>
```

Comment (dextermb) [reply src/DynamicDataTable.jsx]: @DivineOmega but then you won't have margin between the message and indicator. How I have done it is to keep the original margin between the two (or non at all if only one is passed).

Comment (DivineOmega) [reply src/DynamicDataTable.jsx]: @dextermb My thought is that if the user wants a margin between the two, they will be able to specify that in the `loadingIndicator` component instead?

File: README.md [hunk 4]
```diff
@@ -319,16 +319,24 @@ By default, the React Dynamic Data Table will not show indication that it is
...
+pass a React component into the `loadingIndicator` prop, which will be displayed
+above the textual loading message.
 
-If you wish to replace the entire table while data is being loaded in then you
-can use the `loadingComponent` prop to pass a valid react element to be displayed
-instead of the default loading message.
+```JSX
+<DynamicDataTable
+    loading={true}
+    loadingMessage={'User data is now loading...'}
```
Comment (dextermb) [inline README.md]: I believe that our style guide says to use `"` (double quotes) for strings

File: src/DynamicDataTable.jsx (hunk 1 above)
Comment (dextermb) [reply src/DynamicDataTable.jsx]: > @dextermb My thought is that if the user wants a margin between the two, they will be able to specify that in the `loadingIndicator` component instead?

Ah definitely a good suggestion! Let's do that then.

File: README.md (hunk 4 above)
Comment (DivineOmega) [reply README.md]: Good point. Resolved.

**Pull request merged**